introduction.Rmd

```diff
@@ -184,6 +184,309 @@ We call the combination of these two steps the __quote and unquote__ pattern. Th
 Most of the time though, we don't need to create blueprints manually. We'll get them by quoting the arguments supplied by users. This gives your functions the same usage and feel as tidyverse verbs.
 
 
+# Do you need tidy eval? {#sec:do-you-need}
+
+In the field of computer science, frameworks like tidy evaluation are known as metaprogramming. Modifying the blueprints of computations amounts to programming the program, i.e. metaprogramming. In other languages, this type of approach is often seen as a last resort because it requires new skills and might make your code harder to read. Things are different in R because of the importance of data masking functions, but it is still good advice to consider other options before turning to tidy evaluation. In this section, we review several strategies for solving programming problems with tidyverse packages.
+
+Before diving into tidy eval, make sure to know about the fundamentals of programming with the tidyverse:
+
+* [Fixed column names](#sec:fixed-colnames). A solid function taking data frames with fixed column names is better than a brittle function that uses tidy eval.
+
+* [Automating loops](#sec:automating-loops). dplyr excels at automating loops. Acquiring a good command of rowwise vectorisation and columnwise mapping may prove very useful.
+
+Tidy evaluation does not have to be hard. It features several lightweight techniques that you can easily pick up in your workflow:
```

Comment:
Tidy eval light? Or maybe something to the effect of "Using tidy evaluation does not have to be all-or-nothing."? (Also thinking about how to get the words _hard_ and _easy_ out, since they're relative).

Reply:
Perhaps:

Tidy evaluation is not all-or-nothing, it encompasses a wide range of features and techniques. Here are a few techniques that are easy to pick up in your workflow: